docs/index.asciidoc

```diff
@@ -61,7 +61,9 @@ send them to the logstash server.
   * Value type is <<array,array>>
   * There is no default value for this setting.
 
-list of addresses lumberjack can send to
+List of addresses lumberjack can send to. When the plugins needs to connect to the remote
+peer it randomly selects one of the hosts. The connection happens when the plugin is registered 
+and when it detects a connection error.
```

Comment:
```suggestion
List of addresses lumberjack can send to. When the plugin needs to connect to the remote
peer, it randomly selects one of the hosts. The connection happens when the plugin is registered 
and when it detects a connection error.
```

Reply:
> "The connection happens when the plugin is registered 
and when it detects a connection error."

This sentence seems counter intuitive.  What is it about the connection error that makes the connection happen?  

Maybe if one connection fails, it moves to another one?

Reply:
Thanks @karen to look at this, I try to explain the logic flow. When the pipeline is created and the plugin registered it open a connection to one of the hosts. When the plugin is sending data and receives an IO error (due to network or host problems) is opens a freshly connection, bringing the new host from the `hosts` list. How the plugin selects the host to connect, is always in a random way.

Reply:
Gotcha! Great explanation.
Let's make it clear that we're covering two different scenarios--happy path and failure.  

What about something like: 
>When the plugin is registered, it opens a connection to one of the hosts.
If the plugin detects a connection error, it selects a different host from the list and opens a new connection.